docs/sources/mimir/release-notes/v2.12.md

```diff
@@ -0,0 +1,122 @@
+- **Spread-minimizing token-related CLI flags**: `-ingester.ring.token-generation-strategy`, `-ingester.ring.spread-minimizing-zones` and `-ingester.ring.spread-minimizing-join-ring-in-order`.
+
+## Experimental features
+
+Grafana Mimir 2.12 includes new features that are considered experimental and disabled by default. Please use them with caution and report any issue you encounter:
+
+- **The maximum number of tenant IDs that may be for a federated query** can be configured via the `-tenant-federation.max-tenants` CLI flag on query-frontends. By default, it is 0, meaning that the limit is disabled.
+- **Sharding of active series queries can be enabled** via the `-query-frontend.shard-active-series-queries` CLI flag on query-frontends.
+- **Timely head compaction** can be enabled via the `-blocks-storage.tsdb.timely-head-compaction-enabled` on ingesters. If enabled, the head compaction happens when the min block range can no longer be appended, without requiring 1.5x the chunk range worth of data in the head.
+- **Streaming of responses from querier to query-frontend** can be enabled via the `-querier.response-streaming-enabled` CLI flag on queriers. This is currently supported only for responses from the `/api/v1/cardinality/active_series` endpoint.
+- **The maximum response size for active series queries** (in bytes) can be set via the `-querier.active-series-results-max-size-bytes` CLI flag on queriers.
+- **Metric relabeling on a per-tenant basis can be forcefully disabled** via the `-distributor.metric-relabeling-enabled` CLI flag on rulers. Metrics relabeling is enabled by default.
+
+### Helm chart improvements
+
+The Grafana Mimir and Grafana Enterprise Metrics Helm chart is now released independently. See the [Grafana Mimir Helm chart documentation](/docs/helm-charts/mimir-distributed/latest/).
+
+## Important changes
+
+In Grafana Mimir 2.12 the following behavior has changed:
+
+- Store-gateway now persists a sparse version of the index-header to disk on construction and loads sparse index-headers from disk instead of the whole index-header.
+- Alertmanager deprecated the `v1` API. All `v1` API endpoints now respond with a JSON deprecation notice and a status code of `410`. All endpoints have a `v2` equivalent. The list of endpoints is:
+  - `<alertmanager-web.external-url>/api/v1/alerts`
+  - `<alertmanager-web.external-url>/api/v1/receivers`
+  - `<alertmanager-web.external-url>/api/v1/silence/{id}`
+  - `<alertmanager-web.external-url>/api/v1/silences`
+  - `<alertmanager-web.external-url>/api/v1/status`
+- Exemplar's label `traceID` has been changed to `trace_id`, in order to be consistent with the OpenTelemetry standard.
+- Errors returned by ingesters now contain only gRPC status codes. Previously they contained both gRPC and HTTP status codes. Note: in order to guarantee backwards compatibility when migrating from a version prior to `2.11`, it is necessary to first migrate to version `2.11`, and then to version `2.12`.
```

Comment:
I think this is the most-important item on this list, and should be very prominent. What exactly happens when someone skips few versions? Eg. someone migrates from Mimir 2.8 to 2.12?

Reply:
@pstibrany I have added the requested explanation. WDYT?

Reply:
> Otherwise, it might happen that during the migration, some ingester pre-`2.11` errors with HTTP status code `4xx` will not be recognized as such, and the corresponding request will be repeated.

Thank you, sounds good. If I understand correctly, such requests could end with 5xx error instead of 4xx, right?

Reply:
Yes, but only during the migration if it happens that there is a distributor still running on the old version (<2.11) while ingester is already running on a new version (>=2.12)